### PR TITLE
[GlobalISel][AArch64] Legalize G_EXTRACT_SUBVECTOR

### PR DESCRIPTION
**Pull request by tschuett**
for future combines

### CODE REVIEW

File: llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp [hunk 1]
```diff
@@ -1363,6 +1363,11 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
 
   getActionDefinitionsBuilder({G_SCMP, G_UCMP}).lower();
 
+  getActionDefinitionsBuilder(G_EXTRACT_SUBVECTOR)
+      .legalFor({{v8s8, v16s8}, {v4s16, v8s16}, {v2s32, v4s32}})
+      .widenScalarOrEltToNextPow2(0)
+      .immIdx(0); // Inform verifier imm idx 0 is handled.
```
Comment (arsenm) [inline llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp:1369]: immIdx won't d anything here, just remove it

Comment (tschuett) [reply llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp:1369]: I get verification errors before finding `immIdx`. G_MEMCPY and family also use it.

Comment (arsenm) [reply llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp:1369]: Verification errors? Do you mean the covered rule log?

Comment (tschuett) [reply llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp:1369]:
```
 getLegacyLegalizerInfo().computeTables();
  verify(*ST.getInstrInfo());
```
Yes.

Comment (tschuett) [reply llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp:1369]: In the log was `FAIL` which was supposed to be `OK` for G_EXTRACT_SUBVECTOR.

Comment (arsenm) [reply llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp:1369]: I'd consider that to be a bug in the verify

**Review by arsenm — APPROVED** [review]

**Pull request merged**